src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2008,6 +2036,7 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
             IMethodSymbol debugEntryPoint = null,
             Stream sourceLinkStream = null,
             IEnumerable<EmbeddedText> embeddedTexts = null,
+            Stream metadataPeStream = null,
```

Comment:
What other kinds of metadata-only assemblies are there? Perhaps metadataOnlyAssemblyStream?